util/index.js

```diff
@@ -0,0 +1,191 @@
+*
+*   @raturn {Object} The object containing the config.
+* */
+util.getConfig = function() {
+    let configPath = global.configPath;
+    return fsExtra.readJSONAsync(configPath)
+        .then((config) => {
+            config.appiumPath = config.appiumPath || "appium";
+            config.javaPath = config.appiumPath || "java";
+            return config
+    })
+};
+
+/*
+*   Creates a directory for the output of the current session.
+*   The folder is created by username in STF, a modified ISO time string and uses fs.mkdtempAsync()
+*   to add a random String to the end, if multiple test runs are started at the same timestamp.
+*   (To provide a unique folder name)
+*
+*   @param {String} path The output path given in the config file.
+*   @param {String} userName The username of the current user in STF.
+* */
+util.createSessionDir = function(path, userName) {
+    let basePath = path;
+    let date = new Date().toISOString();
+    date = date.replace(new RegExp(/[:.]/, 'g'), "-");
+    let runPath = basePath + '/' + userName + '_session_' + date;
+
+    fsExtra.ensureDirSync(basePath);
+    return fs.mkdtempAsync(runPath + "-")
+        .then((sessionDir) => {
+            return sessionDir
+    })
+};
+
+/*
+*   Creates a directory for a device.
+*
+*   @param {String} sessionFolder The folder from util.createSessionDir.
+*   @param {String} serial The serial number of the device.
+*
+*   @return {String} The path for the device.
+* */
+util.createDeviceDirectoryStructure = function(sessionFolder, serial) {
+    let devicePath = sessionFolder + "/" + serial;
+    return fs.mkdirAsync(devicePath)
+        .then((err) => {
+            if (err) {
+                console.log(err);
+                return
+            }
+            fs.mkdirAsync(devicePath + "/log").then((err) => { if (err) console.log(err) });
+
+            return devicePath
+        })
+};
+
+/*
+*   Uses portastic to find free ports in a port range.
+*   Seems to work only reliable on Linux.
+*
+*   @param {number} numberOfFreePorts The needed quantity of free ports.
+*   @param {number} minPort The minimum port number.
+*   @param {number} maxPort The maximum port number.
+*
+*   @return {array} An array containing the list of free ports.
+* */
+util.getPortList = function(numberOfFreePorts, minPort, maxPort) {
+    if(minPort > maxPort || maxPort - minPort < numberOfFreePorts){     // check for correct port range because of known bugs in portastic
+        throw { name: "PortRangeError", message: "Wrong port range"}
+    }
+    return portastic.find({
+        min: minPort,
+        max: maxPort,
+        retrieve: numberOfFreePorts
+    }).then((ports) => {
+        return ports;
+    })
+};
+
+/*
+*   Executes the "adb connect" command.
+*
+*   @param {String} remoteConnectUrl The remoteConnectUrl from STF.
+*
+*   @return {boolean} Returns true, is command was successful.
+* */
+util.connectAdbDevice = function(remoteConnectUrl) {
```

Comment:
Maybe it would make sense to use [adbkit](https://github.com/openstf/adbkit) here in the future. 
I just discovered last week that it exists. But maybe it's too much overhead for just the `adb connect` and `adb disconnect` commands here and below...